src/main/java/com/moneytransfer/aspect/IdempotentTransferAspect.java

```diff
@@ -0,0 +1,162 @@
+import org.aspectj.lang.annotation.Around;
+import org.aspectj.lang.annotation.Aspect;
+import org.springframework.stereotype.Component;
+import org.springframework.transaction.annotation.Isolation;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.Optional;
+import java.util.UUID;
+
+@Aspect
+@Component
+@RequiredArgsConstructor
+public class IdempotentTransferAspect {
+    /**
+     * The GetAccount service.
+     */
+    private final GetAccountService getAccountService;
+    /**
+     * The Request service.
+     */
+    private final RequestService requestService;
+    /**
+     * The transaction repository.
+     */
+    private final TransactionRepository transactionRepository;
+
+    /**
+     * Processes the idempotent transfer request.
+     * @param proceedingJoinPoint
+     * @param idempotentTransferRequest
+     * @param requestId
+     * @param newTransferDto
+     * @return a (successful) Transaction
+     * @throws Throwable
+     */
+    @Around("@annotation(idempotentTransferRequest) && execution(* transfer(java.util.UUID, com.moneytransfer.dto.NewTransferDto,..)) && args(requestId, newTransferDto,..)")
+    @Transactional(isolation = Isolation.SERIALIZABLE)
+    public Transaction handleIdempotentTransferRequest(ProceedingJoinPoint proceedingJoinPoint, IdempotentTransferRequest idempotentTransferRequest, UUID requestId, NewTransferDto newTransferDto) throws Throwable {
+        return processRequest(requestId, newTransferDto, proceedingJoinPoint);
+    }
+
+    /**
+     * Processes an idempotent transfer request.
+     *
+     * @param requestId
+     * @param newTransferDto
+     * @param proceedingJoinPoint
+     * @return
+     * @throws Throwable
+     */
+    private Transaction processRequest(final UUID requestId, final NewTransferDto newTransferDto, final ProceedingJoinPoint proceedingJoinPoint) throws Throwable {
+        try {
+            var request = getOrSubmitRequest(requestId, newTransferDto);
+            validateIdempotent(request, newTransferDto);
+            return switch (request.getRequestStatus()) {
+                case SUBMITTED -> processTransfer(request, proceedingJoinPoint);
+                case RESOLVED -> retrieveTransaction(request);
+            };
+        } catch (RuntimeException e) {
+            getOrSubmitRequest(requestId, newTransferDto);
+            throw e;
+        }
+    }
+
+    /**
+     * Gets or submits the request.
+     *
+     * @param requestId
+     * @param newTransferDto
+     * @return a submitted Request
+     * @throws MoneyTransferException
+     */
+    private Request getOrSubmitRequest(final UUID requestId, final NewTransferDto newTransferDto) throws MoneyTransferException {
+        Optional <Request> optionalRequest = Optional.ofNullable(requestService.getRequest(requestId));
+        return optionalRequest.isPresent() ? optionalRequest.get() : requestService.submitRequest(requestId,newTransferDto);
```

Comment:
i think that there's an orElse method at Optional interface

Reply:
ok. I used orElseGet()